Documentation/Signoffs/Microcontroller.md

```diff
@@ -0,0 +1,50 @@
+
+These jobs require a sufficiently powerful board.  One that has ample storage, plenty of physical space, decent processing power, sufficient support, and none restrictive.  Before choosing components and detailing the design, the specific constraints of the subsystem need to be known and laid out.  Then any potential solutions can be compared to the constraints to test viability and fulfillment of the subsystem’s purpose. 
+
+## CONSTRAINTS ON THE SUBSYSTEM
+The constraints given below are generalized to allow for comparison of any potential contender for the microcontroller.  Constraints on the subsystem are as follows:
+1. The chosen microcontroller should cost between $20-$60.  This range is gathered from estimating the costs of other components and viewing prices of avaiable options that meet the needs of the subsystem.
+2. The microcontroller must support communication between components.  To do this, Serial Peripheral Interface (SPI), Universal Asynchronous Receiver/Transmitter (UART) as a standard in hardware communication for liaising will need to be achievable on the board. 
+3. External storage support is required as per the shall statements. Here, the most preferred standard for external storage is a Universal Serial Bus (USB) Flash Drive as it is common in households and easily available.
+4. The board must have at least 15 digital input pins and 15 analog input pins. A major goal of the project is to allow for synchronous operations of at least 3 air quality sensors.  Sensors may be either analog or digital, so there must be support for both.  The board must have enough Input/Output (I/O) pins to operate any combination of 3 sensors whether they be digital or analog. Different sensors interface with a potentially different number of pins.  Preliminary research shows that a sensor may take up to 5 pins to operate. Other sensors may have more, but it is more common for 5 and under.  
+5. The cpu clock speed does not need to be very fast as the code will be mostly Input/Output heavy with less computation.  This means that there will be a lot of waiting for responses and does not require a high-speed processor. A cpu clock speed of at least 1 MHz is preferred for fast bootup.  Cpu's over 1 MHz are plentiful in availability and will provide a reliable speed for what computation is present.
+6. There must be sufficient memory for the code. Comparing similar operating code with libraries and compilation show a range of 15 KB above or below 150 KB (Kilobytes) in flash memory.  To account for worst case, 165 KB is the required memory.
```

Comment:
How was this similar code identified? What does it mean to be similar and why are you sure that the actual code will take up less space?